### ISSUE TITLE
Can't get the component to register in nuxt.js both manually copying it or importing

### ISSUE THREAD

**Issue by toddpadwick** [AlbanCrepel/vue-displacement-slideshow#35]
I have installed the plugin via NPM. I have registered the plugin in the plugins directory as follows:

*vue-displacement-slideshow.client.js*
```
import Vue from 'vue';
import VueDisplacementSlideshow from "vue-displacement-slideshow";
console.log(VueDisplacementSlideshow); // this seems to return the component as expected but i just can't use the component anywhere on my site
Vue.use(VueDisplacementSlideshow);

```
I have also imported into nuxt like so:

*nuxt.config.js*

```
plugins: [
    '~/plugins/vue-displacement-slideshow.client.js',
  ],
```

I received the `unknown custom component` error when I tried to use `<vue-displacement-slideshow/>`

After failing, I then tried instead, to copy over the component manually, (while keeping the dependency in package.json so to keep the plugins third party dependencies installed as required (such as three.js). However, this failed as I am now getting the following errors:
``` 
 Unexpected token '!'

```
and on the back end:

```

These dependencies were not found:                                                                                                                                          friendly-errors 18:41:28
                                                                                                                                                                            friendly-errors 18:41:28
* three/src/cameras/OrthographicCamera.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/constants.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/geometries/PlaneGeometry.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/loaders/TextureLoader.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/materials/ShaderMaterial.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/math/Vector2.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/objects/Mesh.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/renderers/WebGLRenderer.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/scenes/Scene.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&
* three/src/textures/Texture.js in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./components/VueDisplacementSlideshow.vue?vue&type=script&lang=js&

```

I have `"vue-displacement-slideshow": "^3.0.1"`  in my package.json

Any ideas what I am doing wrong here?

**Comment by AlbanCrepel** [comment]
Hi, I got it to work on a Nuxt project by just wrapping the component with a `<no-ssr>` tag, like explained in the README:
```vue
<no-ssr>
    <vue-displacement-slideshow />
</no-ssr>
```

And in your script tag:
```js
import VueDisplacementSlideshow from "vue-displacement-slideshow";

export default {
    components: {VueDisplacementSlideshow}
}
```

If you still want to use it as a plugin, something like this should work:
```
// plugins/vueDisplacementSlideshow.js
import Vue from 'vue'
import vDisplacementSlideshow from 'vue-displacement-slideshow';

const VueDisplacementSlideshow = {
    install(Vue, options) {
        Vue.component('vue-displacement-slideshow', vDisplacementSlideshow)
    }
};
Vue.use(VueDisplacementSlideshow);
```

And add this object in the `plugins` array in `nuxt.config.js` :
```js
{ src: '@/plugins/vueDisplacementSlideshow', ssr: false }
```

Let me know if that worked for you :)

**Comment by toddpadwick** [comment]
Excellent, thanks – I got it working. The issue was, as you say, the `<no-ssr>` wrapper –  I did see this in the docs but as `no-ssr` is depreciated, now replaced by <client-only> I simply used client-only as I thought it was exactly the same thing. However, the component works with`no-ssr` works but not with `client-only`.

This may become an issue later on when Nuxt drops support for no-ssr completely in v3.
